Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -5189,6 +5189,80 @@ bool CStaticFunctionDefinitions::GUIComboBoxSetItemText(CClientEntity& Entity, i
     return false;
 }
 
+int CStaticFunctionDefinitions::GUIComboBoxGetItemCount(CClientEntity& Entity)
+{
+    RUN_CHILDREN(GUIComboBoxGetItemCount(**iter))
+
+    // Are we a CGUI Element?
+        if (IS_GUI(&Entity))
+        {
+            CClientGUIElement& GUIElement = static_cast<CClientGUIElement&>(Entity);
+
+            // Are we a combobox?
+            if (IS_CGUIELEMENT_COMBOBOX(&GUIElement))
+            {
+                // Call getItemCount
+                return static_cast<CGUIComboBox*>(GUIElement.GetCGUIElement())->GetItemCount();
+        }
+    }
+    return 0;
+}
+
+bool CStaticFunctionDefinitions::GUIComboBoxSetState(CClientEntity& Entity, int state)
+{
+    RUN_CHILDREN(GUIComboBoxSetState(**iter, state))
+
+    // Are we a CGUI Element?
+        if (IS_GUI(&Entity))
+        {
+            CClientGUIElement& GUIElement = static_cast<CClientGUIElement&>(Entity);
+
+            // Are we a combobox? and state > 0 and state < 3
+            if (IS_CGUIELEMENT_COMBOBOX(&GUIElement) && state > 0 && state < 3)
+            {
+                if (state == 1)
+                {
+                    static_cast<CGUIComboBox*>(GUIElement.GetCGUIElement())->ShowDropList();
+                    return true;
+                }
+                else if (state == 2)
+                {
+                    static_cast<CGUIComboBox*>(GUIElement.GetCGUIElement())->HideDropList();
+                    return true;
+                }
+            }
+        }
+    return false;
+}
+
+int CStaticFunctionDefinitions::GUIComboBoxGetState(CClientEntity& Entity)
+{
+    RUN_CHILDREN(GUIComboBoxGetState(**iter))
+
+        // Are we a CGUI Element?
+        if (IS_GUI(&Entity))
+        {
+            CClientGUIElement& GUIElement = static_cast<CClientGUIElement&>(Entity);
+            int comboState;

```

Comment:
Tip: Try reducing the scope of variables as much as possible, i.e. move `comboState` into the if block or avoid creating it entirely by directly returning from the `vs` / `!vs` - if blocks.